automation/src/test/scala/org/broadinstitute/dsde/workbench/WebBrowserSpec.scala

```diff
@@ -35,17 +30,43 @@ trait WebBrowserSpec extends WebBrowserUtil with ExceptionHandling with LazyLogg
     * @param testCode the test code to run
     */
   def withWebDriver(testCode: (WebDriver) => Any): Unit = {
+    withWebDriver(System.getProperty("java.io.tmpdir"))(testCode)
+  }
+
+  /**
+    * Executes a test in a fixture with a managed WebDriver. A test that uses
+    * this will get its own WebDriver instance will be destroyed when the test
+    * is complete. This encourages test case isolation.
+    *
+    * @param downloadPath a directory where downloads should be saved
+    * @param testCode the test code to run
+    */
+  def withWebDriver(downloadPath: String)(testCode: (WebDriver) => Any): Unit = {
+    val capabilities = getChromeIncognitoOption(downloadPath)
     val headless = new SystemProperties().get("headless")
     headless match {
-      case Some("false") => runLocalChrome(testCode)
-      case _ => runHeadless(testCode)
+      case Some("false") => runLocalChrome(capabilities, testCode)
+      case _ => runHeadless(capabilities, testCode)
     }
   }
 
-  private def runLocalChrome(testCode: (WebDriver) => Any) = {
-    val service = new ChromeDriverService.Builder().usingDriverExecutable(new File(FireCloudConfig.ChromeSettings.chromDriverPath)).usingAnyFreePort().build()
+  private def getChromeIncognitoOption(downloadPath: String): DesiredCapabilities = {
+    val options = new ChromeOptions
+    options.addArguments("--incognito")
```

Comment:
Question: why is it necessary to do the tests in incognito?

Reply:
@breilly2 This is copied from your branch - could you comment?

Reply:
It's really not that importabt tbh

Reply:
There was a sense (and maybe even some evidence?) that suggested it would help with the sign-in challenges. I'm highly skeptical but I don't think I can make an argument for "it's not necessary" that would beat the argument of "it doesn't hurt".